From a94428a7d7fdf0ff10212cb299d87805117f7fab Mon Sep 17 00:00:00 2001
From: Linhao Zhang <linhao.zhang@windriver.com>
Date: Thu, 22 May 2014 09:52:43 +0800
Subject: [PATCH] FIX: lttng build error if !CONFIG_UID16 on arm

when CONFIG_UID16 is not set, the old_uid_t is not defined.
If the macros such as OVERRIDE_32_sys_getuid16 is not defined,
the type old_uid_t will be used in lttng,
so define the macros to avoid that.

Signed-off-by: Linhao Zhang <linhao.zhang@windriver.com>
---
 .../arm-32-syscalls-3.4.25_integers_override.h     |   36 ++++++++++++++++++++
 .../arm-32-syscalls-3.4.25_pointers_override.h     |   19 ++++++++++
 2 files changed, 55 insertions(+), 0 deletions(-)

diff --git a/drivers/staging/lttng2/instrumentation/syscalls/headers/arm-32-syscalls-3.4.25_integers_override.h b/drivers/staging/lttng2/instrumentation/syscalls/headers/arm-32-syscalls-3.4.25_integers_override.h
index 895370f..558f500 100644
--- a/drivers/staging/lttng2/instrumentation/syscalls/headers/arm-32-syscalls-3.4.25_integers_override.h
+++ b/drivers/staging/lttng2/instrumentation/syscalls/headers/arm-32-syscalls-3.4.25_integers_override.h
@@ -5,6 +5,24 @@
 
 #ifndef CREATE_SYSCALL_TABLE
 
+#ifndef CONFIG_UID16
+
+#define OVERRIDE_32_sys_getuid16
+#define OVERRIDE_32_sys_getgid16
+#define OVERRIDE_32_sys_geteuid16
+#define OVERRIDE_32_sys_getegid16
+#define OVERRIDE_32_sys_setuid16
+#define OVERRIDE_32_sys_setgid16
+#define OVERRIDE_32_sys_setfsuid16
+#define OVERRIDE_32_sys_setfsgid16
+#define OVERRIDE_32_sys_setreuid16
+#define OVERRIDE_32_sys_setregid16
+#define OVERRIDE_32_sys_fchown16
+#define OVERRIDE_32_sys_setresuid16
+#define OVERRIDE_32_sys_setresgid16
+
+#endif
+
 SC_TRACE_EVENT(sys_arm_fadvise64_64,
 	TP_PROTO(int fd, int advice, loff_t offset, loff_t len),
 	TP_ARGS(fd, advice, offset, len),
@@ -39,6 +57,24 @@ SC_TRACE_EVENT(sys_sync_file_range2,
 
 #else	/* CREATE_SYSCALL_TABLE */
 
+#ifndef  CONFIG_UID16 
+
+#define OVERRIDE_TABLE_32_sys_getuid16
+#define OVERRIDE_TABLE_32_sys_getgid16
+#define OVERRIDE_TABLE_32_sys_geteuid16
+#define OVERRIDE_TABLE_32_sys_getegid16
+#define OVERRIDE_TABLE_32_sys_setuid16
+#define OVERRIDE_TABLE_32_sys_setgid16
+#define OVERRIDE_TABLE_32_sys_setreuid16
+#define OVERRIDE_TABLE_32_sys_setregid16
+#define OVERRIDE_TABLE_32_sys_fchown16
+#define OVERRIDE_TABLE_32_sys_setfsuid16
+#define OVERRIDE_TABLE_32_sys_setfsgid16
+#define OVERRIDE_TABLE_32_sys_setresuid16
+#define OVERRIDE_TABLE_32_sys_setresgid16
+
+#endif
+
 #define OVVERRIDE_TABLE_32_sys_mmap
 TRACE_SYSCALL_TABLE(sys_mmap, sys_mmap, 90, 6)
 
diff --git a/drivers/staging/lttng2/instrumentation/syscalls/headers/arm-32-syscalls-3.4.25_pointers_override.h b/drivers/staging/lttng2/instrumentation/syscalls/headers/arm-32-syscalls-3.4.25_pointers_override.h
index 65131bb..858eff3 100644
--- a/drivers/staging/lttng2/instrumentation/syscalls/headers/arm-32-syscalls-3.4.25_pointers_override.h
+++ b/drivers/staging/lttng2/instrumentation/syscalls/headers/arm-32-syscalls-3.4.25_pointers_override.h
@@ -4,6 +4,15 @@
 
 #ifndef CREATE_SYSCALL_TABLE
 
+#ifndef CONFIG_UID16
+#define OVERRIDE_32_sys_getgroups16
+#define OVERRIDE_32_sys_setgroups16
+#define OVERRIDE_32_sys_lchown16
+#define OVERRIDE_32_sys_getresuid16
+#define OVERRIDE_32_sys_getresgid16
+#define OVERRIDE_32_sys_chown16
+#endif
+
 SC_TRACE_EVENT(sys_mmap2,
 	TP_PROTO(void *addr, size_t len, int prot,
                  int flags, int fd, off_t pgoff),
@@ -27,6 +36,16 @@ SC_TRACE_EVENT(sys_mmap2,
 
 #else	/* CREATE_SYSCALL_TABLE */
 
+#ifndef CONFIG_UID16
+#define OVERRIDE_TABLE_32_sys_getgroups16
+#define OVERRIDE_TABLE_32_sys_setgroups16
+#define OVERRIDE_TABLE_32_sys_lchown16
+#define OVERRIDE_TABLE_32_sys_getresuid16
+#define OVERRIDE_TABLE_32_sys_getresgid16
+#define OVERRIDE_TABLE_32_sys_chown16
+#endif
+
+
 #define OVERRIDE_TABLE_32_sys_execve
 TRACE_SYSCALL_TABLE(sys_execve, sys_execve, 11, 3)
 #define OVERRIDE_TABLE_32_sys_clone
-- 
1.7.5.4

